### PR TITLE
[FEATURE] Add stratified train_valid_split similar to sklearn.model_selection.train_test_split

### PR DESCRIPTION
**Pull request by colinkyle**
## Description ##
I added the ability to perform a stratified split in train_valid_split

### Essentials ###
- [X] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [X] Changes are complete (i.e. I finished coding on this PR)
- [X] All changes have test coverage - I'm not too familiar with test writing, but if you can direct me to a tutorial I'd be happy to get it done.
- [X] Code is well-documented

### Changes ###
- [X ] stratify option for test_valid_split, test_utils.test_train_valid_split
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
Backwards compatible, the only edge case I can think of is if someone tries to use a float to stratify their data and end up getting non-sense results.

cc @dmlc/gluon-nlp-team

### CODE REVIEW

**Comment by mli** [comment]
Job PR-933/1 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-933/1/index.html

**Comment by colinkyle** [comment]
Thanks for your patience, I don't see any tests for the functions within gluonnlp.data.utils should I create a new file under the unittest folder? or is there already a file I should modify with a test for train_valid_split?
Thanks!

**Comment by mli** [comment]
Job PR-933/2 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-933/2/index.html

**Comment by szha** [comment]
@colinkyle let's put the test in tests/test_utils.py for now. Or if you prefer to create a new file for it we can do that too.

**Comment by mli** [comment]
Job PR-933/3 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-933/3/index.html

**Comment by mli** [comment]
Job PR-933/4 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-933/4/index.html

File: src/gluonnlp/data/utils.py [hunk 1]
```diff
@@ -261,14 +264,47 @@ def train_valid_split(dataset, valid_ratio=0.05):
...
+        return train, valid
+    else:
+        if not isinstance(stratify, list):
+            raise TypeError('stratify should be a list')
+        if not len(stratify) == len(dataset):
+            raise ValueError('stratify should be the same length as num_train')
+
+        classes, digitized = np.unique(stratify, return_inverse=True)
+        n_classes = len(classes)
+        num_class = np.bincount(digitized)
```
Comment (szhengac) [inline src/gluonnlp/data/utils.py]: One problem of using bincount is that `len(num_class) != n_classes` in some cases, e.g., labels = [0,1,2,4] in which 3 is missing.

**Comment by colinkyle** [comment]
Is this going to move forward? I'm not sure where we are with review.

**Comment by szhengac** [comment]
@colinkyle. It will be merged once it passed the CI.

**Comment by colinkyle** [comment]
@szhengac thanks for the update.

**Comment by szhengac** [comment]
@leezu the CI error is related to downloading a dataset. It seems that we have seen it in other pr before?

**Comment by sxjscience** [comment]
I think we should just rebase and merge.

**Comment by mli** [comment]
Job PR-933/16 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-933/16/index.html

**Comment by mli** [comment]
Job PR-933/17 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-933/17/index.html
